src/provider/provider_devdax_memory.c

```diff
@@ -438,58 +431,51 @@ static umf_result_t devdax_open_ipc_handle(void *provider,
         return UMF_RESULT_ERROR_INVALID_ARGUMENT;
     }
 
-    devdax_memory_provider_t *devdax_provider =
-        (devdax_memory_provider_t *)provider;
     devdax_ipc_data_t *devdax_ipc_data = (devdax_ipc_data_t *)providerIpcData;
 
-    // verify it is the same devdax - first verify the path
-    if (strncmp(devdax_ipc_data->dd_path, devdax_provider->path, PATH_MAX)) {
-        LOG_ERR("devdax path mismatch (local: %s, ipc: %s)",
-                devdax_provider->path, devdax_ipc_data->dd_path);
-        return UMF_RESULT_ERROR_INVALID_ARGUMENT;
-    }
-
-    // verify the size of the /dev/dax
-    if (devdax_ipc_data->dd_size != devdax_provider->size) {
-        LOG_ERR("devdax size mismatch (local: %zu, ipc: %zu)",
-                devdax_provider->size, devdax_ipc_data->dd_size);
-        return UMF_RESULT_ERROR_INVALID_ARGUMENT;
-    }
-
-    umf_result_t ret = UMF_RESULT_SUCCESS;
-    int fd = utils_devdax_open(devdax_provider->path);
+    int fd = utils_devdax_open(devdax_ipc_data->path);
     if (fd == -1) {
-        LOG_PERR("opening a devdax (%s) failed", devdax_provider->path);
+        LOG_PERR("opening the devdax (%s) failed", devdax_ipc_data->path);
         return UMF_RESULT_ERROR_INVALID_ARGUMENT;
     }
 
     unsigned map_sync_flag = 0;
     utils_translate_mem_visibility_flag(UMF_MEM_MAP_SYNC, &map_sync_flag);
 
+    // length and offset passed to mmap() have to be page-aligned
+    size_t offset_aligned = devdax_ipc_data->offset;
+    size_t length_aligned = devdax_ipc_data->length;
+    utils_align_ptr_down_size_up((void **)&offset_aligned, &length_aligned,
```

Comment:
Do we really need this call to the `utils_align_ptr_down_size_up` function? Offset and length should be aligned during the allocation? I mean that IPC handle represents allocation that is done by this memory provider and it already should be page-aligned. Probably here and in the `devdax_open_ipc_handle` we need to check that offset and length are page-aligned.

Reply:
Yes, it is needed. `alloc()` aligns pointer's address to the alignment given as the alloc()' parameter, not to the page size,
but `mmap()`requires the page size alignment.

Reply:
Ok, I got it. In case of arbitrary alignment such logic makes sense, but does arbitrary alignment in case of any memory provider (not only devdax) makes sense. I can assume that alignment always is a multiply of the page size. Did I miss something?

Reply:
User-requested alignment (given as `alloc(size, alignment)`) "_must be a power of two and a multiple of sizeof(void *)._" as POSIX_MEMALIGN(3) states.

Reply:
Yeah, but the memory provider API is closer to the `mmap`.
The `umfPoolMalloc()` is similar to regular `malloc`, but memory providers are responsible for a coarse-grain allocations.

Reply:
Tests shows that the /dev/dax device created with the 2MB alignment requires the same 2MB alignment in `mmap()` ...

Reply:
... so it is required here to get the 2MB-alignment.

Reply:
> So do you think that all memory providers should require the alignment to be a multiply of the page size?

I think so, unless any objections.

@bratpiorka @igchor @pbalcer  what is your opinion on that?

Reply:
I'd put it more strongly, not only alignment, but also size, must always be a multiple of a page size. Because no implementation can do anything different, so we always end up allocating more for misaligned values.
If someone is using an interface this low-level, they should be aware of its limitations.

Reply:
I would add that this should be a multiply of provider->get_min_page_size()

Reply:
@vinser52 Done, please resolve.

Reply:
The devdax tests passed: https://github.com/oneapi-src/unified-memory-framework/actions/runs/11551236658

Reply:
But you still call the `utils_align_ptr_down_size_up` function to align the `offset` and `length`.
As we discussed, the  `offset` and `length` should be already aligned. Did I miss something?

Reply:
No, that's me who missed part of the above discussion ;-)

Reply:
@vinser52 It is done now.

Reply:
No, it is not ready yet.
@pbalcer @vinser52 @bratpiorka jemalloc's `arena_extent_alloc()` always requests `alignment=4096`, but the page size of the devdax provider is 2MB, so the devdax provider cannot require the alignment in `devdax_alloc()` to be a multiple of the devdax page size (2MB), because jemalloc will never work with jemalloc.

Reply:
> I'd put it more strongly, not only alignment, but also size, must always be a multiple of a page size. Because no implementation can do anything different, so we always end up allocating more for misaligned values. If someone is using an interface this low-level, they should be aware of its limitations.

@pbalcer @vinser52 @bratpiorka jemalloc's `arena_extent_alloc()` always requests `alignment=4096`, but the page size of the devdax provider is 2MB, so the devdax provider cannot require the alignment in `devdax_alloc()` to be a multiple of the devdax page size (2MB), because jemalloc will never work with the devdax provider.

Reply:
I reverted to the previous version. This version works correctly.
Done.

Reply:
Regarding jemalloc, it is interesting. Do you know how jemalloc work with huge pages?

Reply:
Well, 2mb is aligned to 4096. So in this case it's safe to align-up to pagesize.

Jemalloc should support huge pages, so I'm surprised as well. But it probably depends on the specific allocations being made...

Reply:
> Do you know how jemalloc work with huge pages?

No.

Reply:
I think the main issue is that we cannot require the size (in `alloc()`) to be a multiple of 2MB - we cannot force jemalloc or any other pool manager or a user of a memory provider to call `alloc()` only with size being a multiple of 2MB.

Reply:
For example:
```
[ RUN      ] DevDaxProviderProxyPoolTest/umfIpcTest.BasicFlow/0
[PID:7956   TID:7956   ERROR UMF] utils_mmap_file: mapping file with the MAP_SYNC flag failed (fd=3, offset=0, length=1054867456): Operation not supported
[PID:7956   TID:7956   DEBUG UMF] utils_mmap_file: file mapped with the MAP_SHARED flag (fd=3, offset=0, length=1054867456)
[PID:7956   TID:7956   DEBUG UMF] devdax_initialize: devdax memory mapped (path=/tmp/devdax, size=1054867456, addr=0x7f680e000000)
[PID:7956   TID:7956   DEBUG UMF] umfTrackingMemoryProviderCreate: upstream=0x7f684d06a168, tracker=0x7f684d078068, pool=0x7f684d073068, ipcCache=0x7f684d065008
[PID:7956   TID:7956   INFO  UMF] umfPoolCreateInternal: Memory pool created: 0x7f684d073068
[PID:7956   TID:7956   DEBUG UMF] umfMemoryTrackerAdd: memory region is added, tracker=0x7f684d078068, ptr=0x7f680e000000, size=400
[PID:7956   TID:7956   ERROR UMF] devdax_open_ipc_handle: incorrect length (400) in IPC handle - it is not page-aligned
[PID:7956   TID:7956   ERROR UMF] trackingOpenIpcHandle: upstream provider failed to open IPC handle
[PID:7956   TID:7956   ERROR UMF] umfOpenIPCHandle: memory provider failed to open the IPC handle.
/home/ldorau/work/unified-memory-framework/test/ipcFixtures.hpp:213: Failure
Expected equality of these values:
  ret
    Which is: 3
  UMF_RESULT_SUCCESS
    Which is: 0
```
(look for: **devdax_open_ipc_handle: incorrect length (400) in IPC handle - it is not page-aligned**)

Reply:
I've just pushed out a version with alignment up of size - just for yours today discussion ;-)

Reply:
Updated

Reply:
`devdax_open_ipc_handle()` contains: 
```
    size_t length_aligned = ALIGN_UP(devdax_ipc_data->length, DEVDAX_PAGE_SIZE_2MB);
```

Reply:
@vinser52 what do you think about the current version of this PR ?

Reply:
@vinser52 please review the new version - there is no up-alignment in open/close_IPC_handle any more

Reply:
The new IPC tests pass too: https://github.com/oneapi-src/unified-memory-framework/actions/runs/11596112796/job/32286377438
See: https://github.com/oneapi-src/unified-memory-framework/pull/845

Reply:
@vinser52 Done.